src/schema-types/string.schema-type.ts

```diff
@@ -1,3 +1,10 @@
+/**
+ * Copyright Websublime All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://websublime.dev/license
+ */
+
```

Comment:
@Hulago should we check for alternative on the regular expression?

Reply:
`
function isValidUrl(string) {  
  try {
    url = new URL(string);
    return true;
  } catch (_) {
    return false;  
  }
}
`

And this alternative code. It also works in node.

https://caniuse.com/url

Reply:
We may need to add additional validation on the protocol.

Reply:
Hmm, protocol is really needed. Wrong things can be passed to this.

<img width="302" alt="image" src="https://user-images.githubusercontent.com/495720/155116458-8695a75b-c658-4bc8-84ca-fbd8cbd0c329.png">

Reply:
This alternative could be the answer. What you think?

```js
const isValidUrl = (url, matchProtocol = false) => {
  const regex = `(http(s)?:\/\/.)${matchProtocol ? '' : '?'}(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)`;
  const expression = new RegExp(regex, 'g');

  return expression.test(url);
};
```

<img width="330" alt="image" src="https://user-images.githubusercontent.com/495720/155119939-b2da9e48-8130-45cb-b639-fb30ff67da68.png">

Reply:
Ok, so we have validations for particular things like protocol way. I agree will be more friendly.

Reply:
Will be applied on PR #5 . Closing to continue.